src/Chapter2.hs

```diff
@@ -841,7 +867,10 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+rotate :: Int -> [a] -> [a]
+rotate n l 
+  | n < 0 = []
+  | otherwise = take (length l) (drop n (cycle l))
```

Comment:
Unfortunately, the standard function `cycle` fails in runtime on empty lists, so you need to handle this case separately.